harness/determined/cli/command.py

```diff
@@ -315,9 +341,28 @@ def parse_config_overrides(config: Dict[str, Any], overrides: Iterable[str]) ->
             if key in _CONFIG_PATHS_COERCE_TO_LIST:
                 value = [value]
 
-        # TODO(#2703): Consider using full JSONPath spec instead of dot
-        # notation.
-        config = _set_nested_config(config, key.split("."), value)
+        # Convert config override in dot notation to json path
+        config_arg_in_json = _dot_to_json(key, value)
+
+        # Some key may have value None of NoneType when value is not defined.
+        # For example, when parent key "slurm" is specified in experiment .yaml file without
+        # any child key, the config will contain None value for key 'slurm':
+        # {'name': 'noop_single', ..., 'entrypoint': 'model_def:NoOpTrial', 'slurm': None}
```

Comment:
Using this line as an example, how do you differentiate between a key, such as 'name', whose value is a simple string, versus a key, such as 'slurm', whose value is a complex structure like the one shown below?

                 slurm:
                   sbatch_args:
                     - --cpus-per-gpu=1

If the user has a config file that has "slurm: None", or "slurm:" with nothing else, wouldn't we just want to issue an error telling the user that key "slurm" requires additional JSON data, and have the user fix it and try again?

Reply:
The example config {'name': 'noop_single', ..., 'entrypoint': 'model_def:NoOpTrial', 'slurm': None} is the internal dictionary representation of the experiment .yaml file, Dict[str, Any]. If the user specifies the "slurm: None" in .yaml file, that will be the 'slurm': 'None', not the NoneType value None. I just tried, creating the experiment will fail too but that is not this PR is about. 

>wouldn't we just want to issue an error telling the user that key "slurm" requires additional YAML data, and have the user fix it and try again?
 Hope @stoksc @ioga  can answer that

Reply:
> If the user specifies the "slurm: None" in .yaml file, that will be the 'slurm': 'None', not the NoneType value None

I suppose in yaml it'd be `slurm: null`.

>wouldn't we just want to issue an error telling the user that key "slurm" requires additional YAML data, and have the user fix it and try again?

the point of this fix is to allow override nested fields in the config from the command line.

Reply:
Perhaps I'm misunderstanding what is taking place in the code, but I don't think my question has been answered.

Here, we are calling "_replace_value()" with the data from the YAML config file, not the config specified in the command line, correct?

```
_replace_value(config, key.split("."), None, {})
```

So, in the example that was given, how do we know which key needs a "{}" for a value and which key it is OK to use "None"?

```
{'name': 'noop_single', ..., 'entrypoint': 'model_def:NoOpTrial', 'slurm': None}
```

Where in the code is the decision being made?

Reply:
_replace_value(config, key.split("."), None, {}) 

It replaces the None value in config with a new value {}

config has the data from the exp YAML file, it may contain None value, for example {...'slurm': None}. Now the user overrides with --config slurm.sbatch_args=[--cpus-per-gpu=1], before we can merge this user overrides with the config, first we fix the {...,'slurm': None,...} in config, change it to {...,'slurm': {},...}. Here key is "slurm.sbatch_args", it is split into a list: [slurm, sbatch_args].

Reply:
But what if the experiment YAML file has this?

```
'name': None
```

With the value for 'name' get replaced with '{}'?

Reply:
No. The key 'name' is not in the overriding key list [slurm, sbatch_args].

Reply:
But what if the experiment config YAML has ```name: None``` and ```name``` is also in the overriding list?

```
det e create --config name="abc" const.yaml .
```

Will name get changed to "{}"?

Reply:
Yes. In this case, you specify name: null in const.yaml, then in config Dict, it has 'name': None, _replace_value will change it to 'name': {}, the next step merge_dicts then will generate the merged config that has 'name': 'abc'

Reply:
It would certainly help if the "Test Plan" section included additional test cases beyond the "slurm:" case to show that these code changes don't have any side effects.